src/hooks/useWalletBalances.ts

```diff
@@ -3,19 +3,77 @@ import { useMemo } from 'react';
 import { Address } from 'viem';
 import useAccountSettings from './useAccountSettings';
 import { useAddysSummary } from '@/resources/summary/summary';
+import { useQueries } from '@tanstack/react-query';
+import { fetchPositions, positionsQueryKey } from '@/resources/defi/PositionsQuery';
+import { RainbowPositions } from '@/resources/defi/types';
+import { add, convertAmountToNativeDisplay } from '@/helpers/utilities';
+import { queryClient } from '@/react-query';
 
-const useWalletBalances = (wallets: AllRainbowWallets) => {
+type WalletBalance = {
+  assetBalanceAmount: number;
+  assetBalanceDisplay: string;
+  positionsBalanceAmount: string | number;
+  positionsBalanceDisplay: string;
+  totalBalanceAmount: string | number;
+  totalBalanceDisplay: string;
+};
+
+type WalletBalanceResult = {
+  balances: Record<Address, WalletBalance>;
+  isLoading: boolean;
+};
+
+const useWalletBalances = (wallets: AllRainbowWallets): WalletBalanceResult => {
   const { nativeCurrency } = useAccountSettings();
 
-  const walletAddresses: Address[] = useMemo(
+  const allAddresses = useMemo(
     () => Object.values(wallets).flatMap(wallet => wallet.addresses.map(account => account.address as Address)),
     [wallets]
   );
 
-  const { data, isLoading } = useAddysSummary({ addresses: walletAddresses, currency: nativeCurrency });
+  const { data: summaryData, isLoading: isSummaryLoading } = useAddysSummary({
+    addresses: allAddresses,
+    currency: nativeCurrency,
+  });
+
+  const positionQueries = useQueries({
+    queries: allAddresses.map(address => ({
+      queryKey: positionsQueryKey({ address, currency: nativeCurrency }),
+      queryFn: () => fetchPositions({ address, currency: nativeCurrency }),
+      enabled: !!address,
+    })),
+  });
+
+  const isLoading = isSummaryLoading || positionQueries.some(query => query.isLoading);
```

Comment:
just want to check:
for the `useAddysSummary` hook, `isLoading` will only be set to `true` at the initial load but will be false afterwards since we have `keepPreviousData` set to `true` which is great.

can we double check that we have something similar for the position queries, and that we aren't going back to `isLoading` set to `true` every time we're fetching after the initial fetch? we should check what we have configured for the positions data

otherwise would expect to result in something like profile balance showing up and then the row disappearing for a bit during a positions refresh, and coming back, etc. also while we're at it, generally check that the data is being refreshed in a reasonable manner now that we rely on this for balances instead of the underlying token data

Reply:
`keepPreviousData` is actually not used correctly there. react query actually "keeps previous data" all the time - even if data is stale, react query still provides it until the new data (fetched in the background) is ready to replace it. `keepPreviousData` keeps the previously fetched data when the query key changes, which is not behavior we want. i'm removing that.

but you're right that the `useWalletBalances` version of `isLoading` is somewhat problematic, for 2 reasons:
1. the addys query has a cache time of 24 hours while the positions queries do not have cache times
2. even if the cache times were the same they would likely not be aligned, especially since these queries could be used in other contexts
this means that the problem you described would be valid every time a user cold starts the app since there is no positions query cache.

the `useWalletBalances` `isLoading` actually isn't used at all though so i can just delete it